packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -175,14 +179,25 @@ export class MonacoWorkspace implements lang.Workspace {
     @inject(ProblemManager)
     protected readonly problems: ProblemManager;
 
+    protected _workspaceFolders: WorkspaceFolder[];
+    get workspaceFolders(): WorkspaceFolder[] {
+        return this._workspaceFolders;
+    }
+
     @postConstruct()
     protected init(): void {
         this.workspaceService.roots.then(roots => {
-            const rootStat = roots[0];
-            if (rootStat) {
-                this._rootUri = rootStat.uri;
-                this.resolveReady();
-            }
+            this._workspaceFolders = roots.map((dir, index) => this.toWorkspaceFolder(dir.uri, index));
+            this.resolveReady();
```

Comment:
Looks like you have changed the behavior. The original code does not resolve if no roots are available. Expected?

Reply:
Good catch. We should also turn `ready` to unresolved when all workspace folders have been removed. I'll prepare that.